### PR TITLE
Update itemlistparser

### PR DESCRIPTION
**Pull request by Raggaer**
This file needs a little update, the current cache setting we use its not needed (the items.xml file is not big enough to impact performance) and makes changing items.xml tedious as you need to reload the parser cache file manually.

This pull request changes the following:

- Remove itemparser.php its not needed anymore
- Remove item_list.txt
- Update itemlistparser.php with the changes and a new method `getItemById($id)`
- We no longer use `ucfirst` when creating the itemlist array. I think users should make use of this (or not).
- Fix conversion from simpleXML objects to array elements (removing ilegal offset warnings).

### CODE REVIEW

**Comment by joseluis2g** [comment]
:laughing:

File: engine/function/itemparser/itemlistparser.php [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
-	$item_list = explode(PHP_EOL, file_get_contents('item_list.txt'));
-	$ia = array();
-	for ($i = 0; $i < count($item_list) - 1; $i++) {
-		$it = explode('@', $item_list[$i]);
-		$ia[(int)$it[0]] = ucfirst($it[1]);
+*/
+
+function getItemList() {	
+	$items = parseItems();	
+	return $items;
```
Comment (Jacobs63) [inline engine/function/itemparser/itemlistparser.php]: Why do we need to assign it to `$items?`

`return parseItems()`

File: engine/function/itemparser/itemlistparser.php [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+
+function parseItems() {
+	global $config;
+
+	$items = simplexml_load_file($config['server_path'].'/data/items/items.xml');
+	$out = array();
+	
+	// Create our parsed item list
+	foreach ($items->children() as $item) {
+		if ($item['id'] && $item['name'] != NULL) {
```
Comment (Jacobs63) [inline engine/function/itemparser/itemlistparser.php]: `!empty($item['id'])`

File: engine/function/itemparser/itemlistparser.php [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
+			$out[(string)$item['id']] = (string)$item['name'];
```
Comment (Jacobs63) [inline engine/function/itemparser/itemlistparser.php]: Why do we need ID to be a string? ID is represented by an integer afaik?
 Will `$item['name']` ever not be a string? Once again, why do we need to retype?

Comment (Raggaer) [reply engine/function/itemparser/itemlistparser.php]: You are right about the id field my bad. And the string conversion is just to stop warnings from using the simple xml object directly as an array item

File: engine/function/itemparser/itemlistparser.php (hunk 1 above)
Comment (Raggaer) [reply engine/function/itemparser/itemlistparser.php]: Updated thanks

File: engine/function/itemparser/itemlistparser.php (hunk 3 above)
Comment (Jacobs63) [reply engine/function/itemparser/itemlistparser.php]: Gotcha, thanks for clarifying.

**Pull request merged**